route-rs-runtime/src/processor/dec_ip_hop.rs

```diff
@@ -54,62 +54,66 @@ impl Processor for DecIpv6HopLimit {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use smoltcp::phy::ChecksumCapabilities;
+    use route_rs_packets::EthernetFrame;
+    use std::convert::TryFrom;
 
     #[test]
     fn test_dec_ipv4_hop_limit() {
         let init_ttl = 64;
-        let repr = Ipv4Repr {
-            src_addr: Ipv4Address::new(10, 0, 0, 1),
-            dst_addr: Ipv4Address::new(10, 0, 0, 2),
-            protocol: IpProtocol::Tcp,
-            payload_len: 10,
-            hop_limit: init_ttl,
-        };
```

Comment:
Oh damn look at that clean constructor. No bytes in sight 👀

Reply:
lol right in the heart.  Yeah the cleaner construction is my next task.  We will have the ability to create "empty" packets that you can populate with all the setters your heart desires.  It is more old school than builder/repr/factory/assemblyline/plzmake/computdo or whatever object rep is all the rage, but it should provide predictable results.

Reply:
Also it comes with the try_froms, encap,decap support you know and love from your favorite packets library

Reply:
_"It's ya BOI"_